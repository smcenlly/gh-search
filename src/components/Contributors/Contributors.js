import React from "react";
import i18n from "i18next";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Contributor } from "../Contributor";

const ContributorsList = styled.ul`
    margin-bottom: 0;
`;

const NoContributions = styled.li`
    list-style: none;
`;

class Contributors extends React.PureComponent {
    static propTypes = {
        contributors: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.object),
            PropTypes.string
        ])
    };

    static NoContributions = NoContributions;

    render() {
        let { className, contributors } = this.props;
        return (
            <ContributorsList className={className}>
                {contributors.length > 0 ? (
                    contributors.map(({ login, contributions, id }) => {
                        return (
                            <Contributor
                                key={id}
                                login={login}
                                contributions={contributions}
                            />
                        );
                    })
                ) : (
                    <NoContributions>
                        {i18n.t("contributors.noContributors")}
                    </NoContributions>
                )}
            </ContributorsList>
        );
    }
}

export default Contributors;
