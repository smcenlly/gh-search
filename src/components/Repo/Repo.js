import React from "react";
import i18n from "i18next";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Contributors } from "../Contributors";
import { colors } from "../../styles";

const RepoWrapper = styled.div`
    padding: 20px;
    & + & {
        border-top: 1px solid ${colors.separator};
    }
`;

class Repo extends React.PureComponent {
    static propTypes = {
        fullName: PropTypes.string,
        contributors: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.object),
            PropTypes.string
        ])
    };
    render() {
        let { className, fullName, contributors } = this.props;
        return (
            <RepoWrapper className={className}>
                <div>{i18n.t("repo.title", { fullName })}</div>
                <Contributors contributors={contributors} />
            </RepoWrapper>
        );
    }
}

export default Repo;
