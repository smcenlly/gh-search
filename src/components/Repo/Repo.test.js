import React from "react";
import { shallow } from "enzyme";
import Repo from "./Repo";
import { Contributors } from "../Contributors";

describe("Repo", function() {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Repo fullName={"fullName"} contirbutors={[]} />);
    });

    it("renders without crashing", () => {
        expect(wrapper).toExist();
    });

    it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("renders contributors with correct props", () => {
        const contributors = [];

        wrapper.setProps({ contributors });

        expect(wrapper).toContainExactlyOneMatchingElement(Contributors);
        expect(wrapper.find(Contributors).first()).toHaveProp({ contributors });
    });
});
