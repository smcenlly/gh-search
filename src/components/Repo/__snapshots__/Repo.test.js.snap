// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo matches snapshot 1`] = `
<styled.div>
  <div />
  <Contributors />
</styled.div>
`;
